### ISSUE TITLE
Sympy returns some of the integral unevaluated

### ISSUE THREAD

**Issue by iamrajiv** [sympy/sympy#15749]
### Sympy returns some of the integral unevaluated

Examples:
`integrate(sqrt(sin(x)))`
`integrate(exp(sin(x)))`
`integrate(sin(x)**Rational(3,2))`

#### for integral `integrate(sqrt(sin(x)))` and `integrate(exp(sin(x)))` it returns

```
>>> from sympy import *
>>> x = symbols('x')
>>> integrate(sqrt(sin(x)))
Integral(sqrt(sin(x)), x)
```

```
>>> from sympy import *
>>> x = symbols('x')
>>> integrate(exp(sin(x)))
Integral(exp(sin(x)), x)
```

Above code is in python 3.7.1 Shell. In Sympy Live it returns in symbolic form but does not evaluate the integral.

#### for integral `integrate(sin(x)**Rational(3,2))` it returns

```
>>> from sympy import *
>>> x = symbols('x')
>>> integrate(sin(x)**Rational(3,2))
Integral(sin(x)**(3/2), x)
```

Above code is in python 3.7.1 Shell. In Sympy Live it gives `Error: Operation timed out.`

There are many more examples of integrals which either return in symbolic form without evaluated or it gives a timeout error. I have run these code in python 3.7.1 Shell, cmd and Sympy Live but none of it works. I think issues #7051 and #15730 is very much related to these issues. Do not know why this is happening but I will try to figure out this.

**Comment by iamrajiv** [comment]
I used Wolfram Alpha to integrate these integrals and it just evaluates it.
#### my approach to this issue:
I think the problem occurs only when we are using the trigonometric function which is raised to some rational power or exponents but if I use `integrate(sin(x)**Rational(4,2))`  which is also have rational power it returns the correct evaluation
```
>>> from sympy import *
>>> x = symbols('x')
>>> integrate(sin(x)**Rational(4,2))
x/2 - sin(x)*cos(x)/2
```
this means that integrals do not evaluate if the power is not an integer though `Rational(4,2))` is also fraction but in simplest form, it becomes an integer i.e. the power of any trigonometric function should not be a fraction provide it should be an integer after expressing it in  simplest form to get integral of that function. I think there may be a problem with the codebase of the integral module of Sympy

I may be wrong please guide on this issue

**Comment by aOrionis** [comment]
The problem here is that there are some integrals that cannot be evaluated yet by Sympy. 
For example the first equation requires an Elliptic integral to similarly some special functions are required to solve the last equation for non-integral exponents. 
This is what I think is happening after some research. Would appreciate some guidance as well.

**Comment by iamrajiv** [comment]
yup @aOrionis Wolfram Alpha also integrates the first example as the elliptic integral of the second kind with parameter `k=mk^2`  and evaluates answer as
![image](https://user-images.githubusercontent.com/42106787/50857310-3f162380-13b3-11e9-87f6-b315a97579a4.png)
The third example is also elliptic integral of the first kind and for second example Wolfram Alpha not able to integrate it. I think maybe `e^sin(x)` is not integrable form but I think there should be a feature in Sympy that if a function is not integrable it should return a message that the function is not integrable.

**Comment by iamrajiv** [comment]
my approach for integration of function `e^sin(x)` is that we should open e^x to Maclaurin series and integrate term by term but it is a long way though

**Comment by iamrajiv** [comment]
I used scipy and numpy to integrate `sqrt(sin(x))` and it gives 
```
>>> from scipy.integrate import quad
>>> import numpy as np
>>> def integrand(x):
	return np.sqrt(np.sin(x))

>>> ans = quad(integrand, 0, np.pi)
>>> print (ans)
(2.396280469471185, 1.91535676208332e-11)
```
I want to just ask that can we convert the answer that we get from above to symbolic form i.e. decimal number to rational number if we could it will be easy to evaluate these integrals
@normalhuman can you guide me on this

**Comment by iamrajiv** [comment]
I used scipy and numpy to integrate `sqrt(sin(x))` and it gives 
```
>>> from scipy.integrate import quad
>>> import numpy as np
>>> def integrand(x):
	return np.sqrt(np.sin(x))

>>> ans = quad(integrand, 0, np.pi)
>>> print (ans)
(2.396280469471185, 1.91535676208332e-11)
```
I want to just ask that can we convert the answer that we get from above to symbolic form i.e. decimal number to rational number if we could it will be easy to evaluate these integrals
@asmeurer @normalhuman can you guide me on this

**Comment by asmeurer** [comment]
If `integrate` returns an `Integral` it means that the algorithms don't know how to integrate the function. The algorithms for integration are heuristical in nature so there will always be functions that have integrals that SymPy cannot find (there is the Risch algorithm, but that only works for elementary integrals, and anyway is only partially implemented in SymPy). 

There's no way to go from a numerical integral to a symbolic one. If you only care about a numerical answer, you may be better off just using scipy.quad from the start. 

The easiest way to add support for an integral that isn't supported is to find a general integration rule that applies to it and add that rule to `manualintegrate`.

**Comment by iamrajiv** [comment]
got it @asmeurer thanks for the guidance...adding general integration rule to `manualintegrate` you mean adding python code to integrate these functions

**Comment by iamrajiv** [comment]
got it @asmeurer thanks for the guidance adding general integration rule to `manualintegrate` you mean adding python code to integrate these functions

**Comment by asmeurer** [comment]
If you do that you should always be careful to verify the answer from nsimplify somehow, e.g., by computing more digits. 

By the way, `Integral(expand(log(x)*apart(1/((x - 1)*(x + 1)))), (x, 0, 1)).doit()` gives `pi**2/8 - I*pi*log(2)`, which I guess is wrong.

**Comment by normalhuman** [comment]
Yes, Meijer G does not always get the right branch of polylog in the antiderivative, but I failed to improve on its present state. #13850

**Comment by sudoer-Huatao** [comment]
I think that using the `evalf()` function you can actually evaluate unevaluated definite integrals by Sympy. For example:
`x = symbols("x")`
`Integral(f, (x, lbound, rbound)).evalf()`
where f is the function you are integrating lbound is the lower bound and rbound is the upper bound
This doesn't return an unevaluated object

**Comment by oscarbenjamin** [comment]
> using the `evalf()` function you can actually evaluate unevaluated definite integrals

This is true if there are no symbols besides the integration variable and it is a single integral (rather than a double/triple/etc integral). It should be better documented (https://github.com/sympy/sympy/issues/23025) and support for multiple integrals should be added (https://github.com/sympy/sympy/issues/21605).
